java/client/src/main/java/glide/api/models/commands/WeightAggregateOptions.java

```diff
@@ -94,4 +124,34 @@ public String[] toArgs() {
             return argumentsList.toArray(new String[0]);
         }
     }
+
+    /**
+     * Represents the mapping of sorted set binary keys to their score weights. Each weight is used to boost
+     * the scores of elements in the corresponding sorted set by multiplying them before their scores
+     * are aggregated.
+     */
+    @RequiredArgsConstructor
+    public static class WeightedKeysBinary implements KeysOrWeightedKeysBinary {
```

Comment:
Can you update existing `WeightedKeys` to store `List<Pair<GlideString, Double>>` (as you have here) and have 2 constructors - one which accepts `GlideString` and second which accepts `String` and converts them?

Reply:
But then should we have two versions of getArgs in the class?

Reply:
Yes it will be `toArgs` and `toBinaryArgs` or so

Reply:
If I change it it won't be inline with other binary implementations